XPlot.DotNet.Interactive.KernelExtensions/XPlotKernelExtension.cs

```diff
@@ -30,24 +31,37 @@ public string GetChartHtml(PlotlyChart chart)
             var scriptNode = document.DocumentNode.SelectSingleNode("//script");
 
             var newHtmlDocument = new HtmlDocument();
+           
             newHtmlDocument.DocumentNode.ChildNodes.Add(divNode);
             newHtmlDocument.DocumentNode.ChildNodes.Add(GetScriptNodeWithRequire(scriptNode));
 
             return newHtmlDocument.DocumentNode.WriteContentTo();
         }
+      
 
         private static HtmlNode GetScriptNodeWithRequire(HtmlNode scriptNode)
         {
             var newScript = new StringBuilder();
-
-            newScript.AppendLine("<script>");
-            newScript.Append(@"
-var xplotRequire = require.config({context:'xplot-2.0.0',paths:{plotly:'https://cdn.plot.ly/plotly-1.49.2.min'}});
-xplotRequire(['plotly'], function(Plotly) {
-");
+            newScript.AppendLine("<script type=\"text/javascript\">");
+            newScript.AppendLine(@"
+var renderPlotly = function() {
+    var xplotRequire = requirejs.config({context:'xplot-2.0.0',paths:{plotly:'https://cdn.plot.ly/plotly-1.49.2.min'}});
+    xplotRequire(['plotly'], function(Plotly) {");
 
             newScript.Append(scriptNode.InnerText);
-            newScript.AppendLine(@"});");
+            newScript.AppendLine(@"});
+};
+if ((typeof(requirejs) !==  typeof(Function)) || (typeof(requirejs.config) !== typeof(Function))) { 
+    var script = document.createElement(""script""); 
+    script.setAttribute(""src"", ""https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js""); 
+    script.onload = function(){
+        renderPlotly();
+    };
+    document.getElementsByTagName(""head"")[0].appendChild(script); 
+}
+else {
+    renderPlotly();
+}");
             newScript.AppendLine("</script>");
```

Comment:
Why not use `PocketView` rather than add a dependency on HtmlAgilityPack?

Reply:
we need parsing anyway to select the div and script element from the xplot html

Reply:
Can't they just be rendered as separate script elements?

Reply:
The script tag injects the other script element conditionally if requrirejs is not defined, they cannot be 2 tags